### PR TITLE
Moving axis calls to sides

### PR DESCRIPTION
**Pull request by jordansread**

### CODE REVIEW

**Comment by jordansread** [comment]
I think you'll like those image diffs @ldecicco-USGS (`las`)

**Comment by ldecicco-USGS** [comment]
:heart_eyes:

File: R/access-gsplot.R [hunk 1]
```diff
@@ -113,8 +113,14 @@ lim <- function(object, side=NULL, axis = NULL, set.undefined=TRUE, if.null=c(0,
...
-  lims <- lapply(side.names, function(x) object[[x]]$lim) %>% 
-    setNames(side.names)
+  lims <- lapply(side.names, function(x) {
+    lim <- object[[x]]$lim
+    if (object[[x]]$reverse){
+      lim <- rev(lim)
+    }
+    return(lim)
+  })
+  names(lims) <- side.names
```
Comment (jordansread) [inline R/access-gsplot.R:123]: This should cover #321 but maybe could be done in a better way ...

File: R/axis.R [hunk 2]
```diff
@@ -71,79 +71,78 @@ axis.gsplot <- function(object, ..., n.minor=0, tcl.minor=0.15, reverse=NULL) {
...
-    arguments1 <- list(arguments = append(list(side=i), user_args[[fun.name]]), 
-                       gs.config=list(n.minor=n.minor, tcl.minor=tcl.minor, 
-                                      reverse=reverse))
-    to.gsplot <- list(arguments1)
-    names(to.gsplot) <- fun.name
+  for(side in sides){
+    # append the side and give it defaults if it doesn't exist
+    object <- modify_side(object, args = list(), side=side) 
     
-    object <- append(object, to.gsplot)
```
Comment (jordansread) [inline R/axis.R:82]: we don't have `add_to_side`, which is really what is happening here, but it isn't just adding it as a new call, it is replacing the default `axis` call that is placed in when you create a new side. I wasn't quite sure how we should do that.

File: R/print.R [hunk 3]
```diff
@@ -88,8 +83,8 @@ print.gsplot <- function(x, ...){
     par(new=TRUE)
   }
   
-  i.axis.noview <- i.axis[which(!defined.sides %in% c(view.info$x, view.info$y))]
-  draw_axis(views, index.axis=i.axis.noview)
+  #i.axis.noview <- i.axis[which(!defined.sides %in% c(view.info$x, view.info$y))]
```
Comment (jordansread) [inline R/print.R:86]: I don't really know why this was here or what purpose it served.

File: R/utils-side.R [hunk 4]
```diff
@@ -222,7 +222,10 @@ add_new_side <- function(object, side.name){
   stopifnot(length(side.name) == 1)
   if (side.name %in% side_names(object))
     stop(side.name, ' already exists, cannot add it.', call. = FALSE)
-  side.template <- list(list(lim = c(NA, NA), log=FALSE, label="", axes = TRUE, reverse = FALSE, usr.lim=c(FALSE, FALSE)))
+  side.template <- list(list(
```
Comment (jordansread) [inline R/utils-side.R:225]: The side template now has a default axis call built in automatically that is overridden if a user calls `axis`

**Comment by jordansread** [comment]
@ldecicco-USGS @aappling-usgs can you build box plot fig with this version? `0.6.1`

**Comment by aappling-usgs** [comment]
sounds good. can i merge this?

**Comment by jordansread** [comment]
I think so 👍

**Pull request merged**